controller/instance_manager_controller.go

```diff
@@ -2534,3 +2540,21 @@ func (m *InstanceManagerMonitor) categorizeProcesses(instanceMap instanceProcess
 func (imc *InstanceManagerController) isResponsibleFor(im *longhorn.InstanceManager) bool {
 	return isControllerResponsibleFor(imc.controllerID, imc.ds, im.Name, im.Spec.NodeID, im.Status.OwnerID)
 }
+
+func isInstanceInTransitionState(spec *longhorn.InstanceSpec, status *longhorn.InstanceStatus) bool {
+	if status.OwnerID == "" || status.CurrentState == "" {
+		return true
+	}
+
+	isInDesiredState := status.CurrentState == spec.DesireState
+
+	isOnDesiredNode := false
+	switch spec.DataEngine {
+	case longhorn.DataEngineTypeV1:
+		isOnDesiredNode = spec.NodeID == "" || spec.NodeID == status.OwnerID
+	case longhorn.DataEngineTypeV2:
+		isOnDesiredNode = spec.NodeID == status.OwnerID
```

Comment:
Explicitly add a comment to explain why v1 and v2 differ and skip the check of `|| spec.NodeID == status.OwnerID`.

Reply:
Why don't we need to check the node ownership?

Reply:
I mean adding a comment to explain why you skip the check `|| spec.NodeID == status.OwnerID` for v2 data engine.

Reply:
Got it, thanks.